### PR TITLE
Working two version

### PR DESCRIPTION
**Pull request by jasiek-stezalski**

### CODE REVIEW

**Comment by jasiek-stezalski** [comment]
@gd-estrepetov, Please review

File: AlgoritmicTask1.iml [hunk 1]
```diff
@@ -1,11 +1,86 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (gd-estrepetov) [inline AlgoritmicTask1.iml]: Idea files should be ignored by Git. What if I am using Eclipse or Netbeans?

File: src/main/java/algorithmic_task/AlgorithmicTaskApplication.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package algorithmic_task;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
+
+@SpringBootApplication
+@EnableAutoConfiguration
```
Comment (gd-estrepetov) [inline src/main/java/algorithmic_task/AlgorithmicTaskApplication.java]: Please find out what annotation SpringBootApplication is doing and optimize this file

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>groupId</groupId>
+    <artifactId>AlgoritmicTask1</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>war</packaging>
```
Comment (gd-estrepetov) [inline pom.xml]: What is purpose of making this packaging as war?

File: src/main/java/algorithmic_task/Task.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
+package algorithmic_task;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class Task {
```
Comment (gd-estrepetov) [inline src/main/java/algorithmic_task/Task.java]: What is the purpose of this class and why it is a bean?

File: src/main/java/algorithmic_task/TaskService.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class TaskService {
+
+    private Task task;
+    private List<Set<String>> dictionary;
```
Comment (gd-estrepetov) [inline src/main/java/algorithmic_task/TaskService.java]: What is the purpose of class Task? Are you sure that it is thread-safe?

Comment (gd-estrepetov) [reply src/main/java/algorithmic_task/TaskService.java]: Let's make dictionary as a separate component which is initialized during start-up.

File: src/test/java/TaskServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        String s1 = "mother in house d sf sdfs f";
+        String s2 = "in house mom";
+        boolean areSimilar = taskService.areSimilar(s1, s2).getResult();
+
+        assertFalse(areSimilar);
+    }
+
+    @Test
+    public void areSimilar_emptySentences_False() {
+
```
Comment (gd-estrepetov) [inline src/test/java/TaskServiceTest.java]: I am not sure that empty sentences are not similar. Could you explain this?

**Review by gd-estrepetov — CHANGES_REQUESTED** [review]
Also don't forget about Javadocs

File: src/test/java/TaskServiceTest.java (hunk 6 above)
Comment (jasiek-stezalski) [reply src/test/java/TaskServiceTest.java]: In my opinion: In instruction of this task is information that sentences are considered as similar, if corresponding words from sentences are in the same set. Empty sentences have no words so they are not similar.

File: AlgoritmicTask1.iml (hunk 1 above)
Comment (jasiek-stezalski) [reply AlgoritmicTask1.iml]: Oh my mistake, I forgot about .gitignore.

File: src/main/java/algorithmic_task/TaskService.java (hunk 5 above)
Comment (jasiek-stezalski) [reply src/main/java/algorithmic_task/TaskService.java]: I created Dictionary class, I think now it is thread safe.

File: src/main/java/algorithmic_task/Task.java (hunk 4 above)
Comment (jasiek-stezalski) [reply src/main/java/algorithmic_task/Task.java]: I created this class to have more information what I got as a result, not only "true", or "false". But now it is not necessary so I remove this class.

**Comment by gd-estrepetov** [comment]
@jasiek-stezalski , let's merge this PR. I will add some issues for you to improve this project.

**Pull request merged**